lib/logger_json/formatters/datadog.ex

```diff
@@ -244,6 +259,29 @@ defmodule LoggerJSON.Formatters.Datadog do
 
   defp format_http_request(_meta), do: nil
 
+  defp format_error(%{message: message} = msg, metadata, level, reported_levels) when is_binary(message) do
+    existing_error = msg[:error]
+
+    if level in reported_levels and is_nil(existing_error) do
```

Comment:
One small last thing, and LGTM. Pattern matching has much better performance, and people already complain about logging being costly :D

```suggestion
  defp format_error(%{message: %{error: _}} = msg, _metadata, _level, _reported_levels), do: nil
  
  defp format_error(%{message: message} = msg, metadata, level, reported_levels) when is_binary(message) do
    if level in reported_levels do
```

Reply:
Done